core-plugins/src/main/java/co/cask/hydrator/plugin/error/ErrorCollector.java

```diff
@@ -50,27 +49,27 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
     if (inputSchema != null) {
       if (config.messageField != null && inputSchema.getField(config.messageField) != null) {
-        throw new IllegalArgumentException(String.format(
+        throw new InvalidConfigPropertyException(String.format(
```

Comment:
Are we using `InvalidConfigPropertyException ` instead of `IllegalArgumentException `?

Reply:
yep, this will allow better UI integration in the future, as it will be able to highlight which field is invalid.

Reply:
I see. should we also change it for other plugins, eg: Joiner, DedupAggregator?

Reply:
Only when it's tied to a specific config property. If it's something about the input schema then it doesn't really make sense.